Extractor/ServerSubscriptionManager.cs

```diff
@@ -0,0 +1,105 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+
+using Cognite.OpcUa.History;
+using Microsoft.Extensions.Logging;
+
+using Opc.Ua;
+using Opc.Ua.Client;
+
+namespace Cognite.OpcUa
+{
+    public class ServerSubscriptionManager
+    {
+        private readonly ILogger<ServerSubscriptionManager> _logger;
+        private readonly UAClient _uaClient;
+        private readonly ServerNamespacesToRebrowseConfig? _config;
+
+        private readonly uint[] attributes = new[]
+        {
+            Attributes.NodeId,
+            Attributes.DisplayName,
+            Attributes.DataType,
+            Attributes.NodeClass,
+            Attributes.Description
+        };
+
+        public ServerSubscriptionManager(
+            ILogger<ServerSubscriptionManager> logger,
+            UAClient uaClient,
+            ServerNamespacesToRebrowseConfig? config
+        )
+        {
+            _logger = logger;
+            _uaClient = uaClient;
+            _config = config;
+        }
+
+        public async Task CreateCustomServerSubscriptions(UAExtractor extractor, CancellationToken token)
+        {
+            var serverNode = await _uaClient.GetServerNode(token);
+
+            List<NodeId> nodeIds = new List<NodeId>();
+            var targets = _config?.NamespaceNodes.Select(name => name.ToNodeId(_uaClient));
+
+            await _uaClient.Browser.BrowseDirectory(
+                new[] { serverNode.Id },
+                (refDef, id) =>
+                {
+                    var nodeId = (NodeId) refDef.NodeId;
+                    if (targets.Contains(nodeId))
+                    {
+                        nodeIds.Add(nodeId);
+                        _logger.LogInformation(
+                            "Subscription to a rebrowse on node {node} with {id} is now set",
+                            refDef.DisplayName.ToString(), nodeId
+                        );
+                    }
+                },
+                token,
+                maxDepth: -1,
+                doFilter: false
+            );
+
+            var readValueIds = new ReadValueIdCollection(
```

Comment:
Do we actually need to read this?

Reply:
I had to do this because I noticed that the `AddSubscriptions` Method takes a list of `UAHistoryExtractionState`.

Reply:
I'm not actually 100% sure it needs to, we should maybe investigate that a bit later.